### PR TITLE
Add well known secret masking

### PR DESCRIPTION
**Pull request by ilya-hontarau**
# Description of the change

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- You have updated the navigation files correctly:
    - [x] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [ ] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/concepts/stack/stack-settings.md [hunk 1]
```diff
@@ -176,6 +176,22 @@ spacectl stack local-preview --id <stack-id>
 
     Use with caution.
 
+### Enable Well Known Masking
+
+Indicates whether secret patterns will be automatically redacted from logs.
+If this is enabled, next secrets will be masked from logs:
```
Comment (dnaploszek) [inline docs/concepts/stack/stack-settings.md:182]:
```suggestion
This setting determines if secret patterns will be automatically redacted from logs. If enabled, the following secrets will be masked from logs:
```

File: docs/concepts/stack/stack-settings.md [hunk 2]
```diff
@@ -176,6 +176,22 @@ spacectl stack local-preview --id <stack-id>
 
     Use with caution.
 
+### Enable Well Known Masking
```
Comment (dnaploszek) [inline docs/concepts/stack/stack-settings.md:179]:
```suggestion
### Enable well known secret masking
```

I think this should use Sentence case

File: docs/concepts/stack/stack-settings.md [hunk 3]
```diff
@@ -176,6 +176,22 @@ spacectl stack local-preview --id <stack-id>
...
+- AWS Access Key Id
+- GitHub PAT
+- GitHub Fine-Grained PAT
+- GitHub App Token
+- GitHub Refresh Token
+- GitHub OAuth Access Token
+- Slack Token
+- PGP Private Key
+- RSA Private Key
+- PKCS#8 Private Key
```
Comment (dnaploszek) [inline docs/concepts/stack/stack-settings.md:193]: Do you think adding regex patterns here makes sense?

Comment (ilya-hontarau) [reply docs/concepts/stack/stack-settings.md:192]: Not sure, regex patterns are details of implementation and can be changed even to not regexp but strings manipulation. Also not sure do they provide any value for the end customer

**Review by dnaploszek — DISMISSED** [review]

**Review by adamconnelly — APPROVED** [review]

File: docs/concepts/stack/stack-settings.md [hunk 4]
```diff
@@ -176,6 +176,21 @@ spacectl stack local-preview --id <stack-id>
...
```
Comment (eliecharra) [inline docs/concepts/stack/stack-settings.md:192]: I this is slightly inaccurate. We match the exchange format here (base64 encoded PEM), we don't really care about the key format.
Maybe we can just write `PEM blocks with BEGIN PRIVATE KEY header` or something like that?

Comment (ilya-hontarau) [reply docs/concepts/stack/stack-settings.md:192]: @eliecharra it makes sense

**Review by eliecharra — APPROVED** [review]

**Review by dnaploszek — APPROVED** [review]

**Pull request merged**